Task 3/Task 3.2/Task 3.2/Task 3.2/DynamicArray.cs

```diff
@@ -0,0 +1,250 @@
+            {
+                array[i] = _myArray[i];
+            }
+            list.CopyTo(array, length);
+            _myArray = array;
+            length += list.Count;
+        }
+
+        public bool Remove(T item)
+        {
+            int index = Array.IndexOf(_myArray, item);
+            if (index == -1)
+            {
+                return false;
+            }
+            else
+            {
+                _myArray[index] = default(T);
+                for (int i = index; i <= length; i++)
+                {
+                    _myArray[i] = _myArray[i + 1];
+                    if (i == length)
+                    {
+                        _myArray[i] = default(T);
+                    }
+                }
+                length--;
+                return true;
+            }
+        }
+
+        public bool Insert(T item, int index)
+        {
+            if (index < 0 || index >= length)
+            {
+                throw new ArgumentOutOfRangeException();
+            }
+            else
+            {
+                if (length < capacity)
+                {
+                    for (int i = length; i >= index; i--)
+                    {
+                        if (i == index)
+                        {
+                            _myArray[index] = item;
+                        }
+                        else
+                        {
+                            _myArray[i] = _myArray[i - 1];
+                        }
+                    }
+                    length++;
+                    return true;
+                }
+                else
+                {
+                    T[] array = new T[capacity * 2];
+                    for (int i = 0; i <= length; i++)
+                    {
+                        if (i < index)
+                            array[i] = _myArray[i];
+                        else if (i == index)
+                            array[i] = item;
+                        else
+                            array[i] = _myArray[i - 1];
+                    }
+                    _myArray = array;
+                    capacity = array.Length;
+                    length++;
+                    return true;
+                }
+            }
+        }
+
+        public void SetCapacity(int value)
+        {
+            if (value >= length)
+            {
+                T[] array = new T[value];
+                for (int i = 0; i < length; i++)
+                {
+                    array[i] = _myArray[i];
+                }
+                _myArray = array;
+                capacity = value;
+            }
+            else
+            {
+                T[] array = new T[value];
+                for (int i = 0; i < array.Length; i++)
+                {
+                    array[i] = _myArray[i];
+                }
+                _myArray = array;
+                capacity = value;
+                length = array.Length;
+            }
+        }
```

Comment:
* You could make a public setter for Capacity property
* array copy code duplication - maybe create shared method?